src/components/ConnectWallet.tsx

```diff
@@ -51,13 +66,35 @@ function ConnectButton({ Tezos, setPublicToken, setWalletConnected }: ButtonProp
 
       // TODO (@srijan): why do I have to cast the wallet twice here?
       Tezos.setWalletProvider(wallet as unknown as WalletProvider);
-
+      setTezos(Tezos);
       walletInfo.setWallet(wallet);
       const activeAccount = await wallet.client.getActiveAccount();
       if (activeAccount) {
         const userAddress = await wallet.getPKH();
         walletInfo.setAddress(userAddress);
       }
+
+      fetch("https://api.granadanet.tzkt.io/v1/contracts/KT1LFf3MEDg4uZCtYHw4RM5zpuJEvF2NPYsJ/storage", {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          if(temporaryHounds.length == 0) {
+            for (let i=0; i < data.counter; i++) {
```

Comment:
I'm not sure if this is well formatted. which also reminds me we should add `pretier` checks to the CI along with ESLint.